src/Disks/IO/CachedReadBufferFromRemoteFS.cpp

```diff
@@ -362,7 +386,7 @@ SeekableReadBufferPtr CachedReadBufferFromRemoteFS::getImplementationBuffer(File
         case ReadType::CACHED:
         {
 #ifndef NDEBUG
-            auto * file_reader = assert_cast<ReadBufferFromFile *>(read_buffer_for_file_segment.get());
+            auto * file_reader = dynamic_cast<ReadBufferFromFileDescriptor *>(read_buffer_for_file_segment.get());
```

Comment:
This.

Now 99% of whole usage of this `CachedBufferFromRemoteFS` is consumed by the `next()` call of the underlying buffer from local fs. See https://pastila.nl/?009d10d1/7098f333e2cc02de544ad7d3409216e9.